task/backend/scheduler.go

```diff
@@ -819,6 +827,7 @@ func (r *runner) updateRunState(qr QueuedRun, s RunStatus, runLogger *zap.Logger
 		r.taskControlService.AddRunLog(r.ts.authCtx, r.task.ID, qr.RunID, time.Now(), "Completed successfully")
 	case RunFail:
 		r.ts.metrics.FinishRun(r.task.ID.String(), false, time.Since(qr.startedAt))
+		r.ts.metrics.LogError(err)
```

Comment:
Given `RunSuccess` can pass an error if `json.Marshal` fails, you will can move this outside the switch and log an error if `err != nil` so that we can pass an error in for any value of `RunStatus`.